lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -1870,12 +1866,21 @@ static bool s1ap_send_enb_deregistered_ind(
     arg->current_ue_index =
         (uint8_t)(arg->handled_ues % S1AP_ITTI_UE_PER_DEREGISTER_MESSAGE);
```

Comment:
Tariq's comment: "This can be implemented without having to use an expensive MOD operation on every call."

You can check if current_ue_index == S1AP_ITTI_UE_PER_DEREGISTER_MESSAGE and reset it to zero.

Reply:
Thanks for pointing to remove MOD operation. The suggested logic here (if current_ue_index == S1AP_ITTI_UE_PER_DEREGISTER_MESSAGE, set it to zero else set it to arg->handled_ues.) would not work because for 2nd iteration onward, the handled_ues count will be above 128 and therefore, we can't set the same to current_ue_index.

However, I have removed the MOD operation from this code segment and restructured the if condition afterwards to get the correct logic. I have updated the PR with the change.

Reply:
yes, my mistake. I meant to increment it.